README.md

```diff
@@ -93,9 +106,279 @@ src/
+```
+
+## Group imports by location
+
+Be organized with your import grouping. Here's the order of my groups:
+
+- NPM packages
+- constants
+- helpers
+- basics
+- components
+- features
+- local imports
+
+## If a constant exists, always import it
+
+Don't use a string literal if there exists a constant for it. It's easier to
+grep for a constant.
+
+```js
+// bad
+const button = <Button size="small" theme="gray" />;
+
+// good
+const button = <Button size={INPUT_SIZE.small} theme={BUTTON_THEME.gray} />;
+```
+
+The known trade-off here is it can be hard to spot small misspellings, so be
+careful!
+
+```js
+// this might not throw an error
+const button = <Button size={INPUT_SIZE.smol} theme={BUTTON_THEME.grey} />;
+```
+
+# Exports
+
+## All files should only export named variables, not defaults
+
+# Components
 
 ## Prefer functions
 
 React components should be functions, not React.Component classes. Stateful
 components should use hooks.
+
+## File organization
+
+Try to organize your files in this order:
+
+- Imports
+- Styled components
+- Constants local to only this file
+- Local functions
+- Exported functions
+- React component
+- `PropType` definitions
+
+# Styling
+
+## Styled components
+
+### Naming
+
+Styled components that are not exported should be suffixed with `El`.
+
+```js
+// bad
+const Text = styled.span``;
+
+// good
+const TextEl = styled.span``;
+
+export const Text = styled.span``;
+```
+
+### ALWAYS merge ThemeProvider themes
+
+By default, the styled-components docs tell you to use ThemeProvider like this:
+
+```
+// bad
+<ThemeProvider theme={{ someThemeProperty: "#000" }}>
+  ...
+</ThemeProvider>
+```
+
+_Do not do this !!_ If you do, it will clobber parent theme properties! At best you will get a lot of errors and at worst
+you'll get a ton of difficult-to-find, difficult-to-debug mismatched colors!
+
+Instead, you may pass a function to `theme` that returns a new theme object. Use
+that to mix in the existing theme with your new stuff:
+
+```
+// v good
+<ThemeProvider theme={(defaultTheme) => ({
+  ...defaultTheme,
+  someThemeProperty: "#000",
+})}>
+  ...
+</ThemeProvider>
+```
+
+# Numbers
+
+## Always use Big or BigNumber for crypto amounts
+
+For all numbers that describe crypto amounts (prices, volumes, etc), store and
+manipulate them as Bigs. Floating point imprecision makes it unsuitable for
+cryptocurrency amounts.
+
+## Always manipulate numbers using Big functions
+
+Use Big's `plus`, `minus`, `div`, and `times` to perform math. Use those
+specific names, don't use their aliases.
+
+## If something needs to be a real `number`, parse it as late as possible
+
+If something like a charting library needs to use real JavaScript numbers, wait
+until the last possible second to parse the Big values. (If possible, do it
+right as you call the function that needs numbers.)
+
+## You don't need to use `toString()`
+
+Referencing a Big number will return a string, and things like Math.min/max will
```

Comment:
IIRC I needed to use `toString()` when rendering Big numbers in React output, it complained about object children otherwise.